### PR TITLE
core/validatorapi: add support for `DutySyncMessage`

### PR DESCRIPTION
**Pull request by xenowits**
Adds support for `DutySyncMessage` to validatorapi component. Also, adds tests.

category: feature
ticket: #1188

### CODE REVIEW

File: core/validatorapi/router_internal_test.go [hunk 1]
```diff
@@ -548,6 +548,23 @@ func TestRouter(t *testing.T) {
 
 		testRouter(t, handler, callback)
 	})
+
+	t.Run("submit_sync_committee_messages", func(t *testing.T) {
+		msgs := []*altair.SyncCommitteeMessage{testutil.RandomSyncCommitteeMessage()}
```
Comment (corverroos) [inline core/validatorapi/router_internal_test.go:553]: submit two to test the slice

File: core/validatorapi/validatorapi_test.go [hunk 2]
```diff
@@ -1321,3 +1321,39 @@ func TestComponent_SubmitAggregateAttestations(t *testing.T) {
...
+
+	vapi.Subscribe(func(_ context.Context, duty core.Duty, set core.ParSignedDataSet) error {
+		require.Equal(t, core.NewSyncMessageDuty(int64(msg.Slot)), duty)
+
+		pk, err := core.PubKeyFromBytes(pubkey[:])
+		require.NoError(t, err)
+
+		data, ok := set[pk]
+		require.True(t, ok)
+		require.Equal(t, core.NewPartialSignedSyncMessage(msg, 0), data)
```
Comment (corverroos) [inline core/validatorapi/validatorapi_test.go:1353]: how do you know this code is actually called? (please do not use ctx cancel pattern, anymore, rather use something like

```
done := make(chan struct)
...
close(done)
...
<-done   
```

**Review by corverroos — APPROVED** [review]

File: core/validatorapi/validatorapi_test.go (hunk 2 above)
Comment (xenowits) [reply core/validatorapi/validatorapi_test.go:1353]: we can add a counter which is incremented each time the subscription function is called. we can use it to assert that this code block is actually called.

Comment (xenowits) [reply core/validatorapi/validatorapi_test.go:1353]: Sure, but this test doesn't use the ctx cancel pattern

Comment (corverroos) [reply core/validatorapi/validatorapi_test.go:1353]: as long as it is threadsafe then sure

**Pull request merged**